### ISSUE TITLE
Doesn't compile on Linux: unknown type name ‘user_ssize_t’

### ISSUE THREAD

**Issue by chrisjbillington** [sickill/stderred#82]
I get the below errors during build on Arch Linux. Looks to be caused by the changes in b401c851f3f15e9a48d1411fc67f97d9f9171802.

Perhaps these types are macos specific?

```C
[ 66%] Building C object CMakeFiles/test_stderred.dir/stderred.c.o
/home/bilbo/.cache/yay/stderred-git/src/stderred/src/stderred.c:387:1: error: unknown type name ‘user_ssize_t’
  387 | user_ssize_t __write_nocancel(int fd, user_addr_t cbuf, user_size_t nbyte);
      | ^~~~~~~~~~~~
/home/bilbo/.cache/yay/stderred-git/src/stderred/src/stderred.c:387:39: error: unknown type name ‘user_addr_t’
  387 | user_ssize_t __write_nocancel(int fd, user_addr_t cbuf, user_size_t nbyte);
      |                                       ^~~~~~~~~~~
/home/bilbo/.cache/yay/stderred-git/src/stderred/src/stderred.c:387:57: error: unknown type name ‘user_size_t’; did you mean ‘ssize_t’?
  387 | user_ssize_t __write_nocancel(int fd, user_addr_t cbuf, user_size_t nbyte);
      |                                                         ^~~~~~~~~~~
      |                                                         ssize_t
/home/bilbo/.cache/yay/stderred-git/src/stderred/src/stderred.c:390:45: error: unknown type name ‘user_addr_t’
  390 | user_ssize_t FUNC(__write_nocancel)(int fd, user_addr_t cbuf, user_size_t nbyte) {
      |                                             ^~~~~~~~~~~
/home/bilbo/.cache/yay/stderred-git/src/stderred/src/stderred.c:390:63: error: unknown type name ‘user_size_t’; did you mean ‘ssize_t’?
  390 | user_ssize_t FUNC(__write_nocancel)(int fd, user_addr_t cbuf, user_size_t nbyte) {
      |                                                               ^~~~~~~~~~~
      |                                                               ssize_t
```

**Comment by AgenttiX** [comment]
Confirmed on Kubuntu 20.04 and 20.10. Could this be fixed by including some headers, or by conditionally defining the types based on the OS? Perhaps automatic CI builds such as GitHub Actions could be used to prevent these kind of issues in the future.

**Comment by chrisjbillington** [comment]
Fixed with a simple:

```C
  #define user_ssize_t ssize_t
  #define user_addr_t const void *
  #define user_size_t size_t
```
on Linux.

See PR #83

**Comment by cehoffman** [comment]
Sorry about this. Checked this time on Linux and macos opting for the more
usual types. I had honestly thought `__write_nocancel` was a Mac specific
implementation. Fixed in b2238f7c

**Comment by chrisjbillington** [comment]
Thanks!
